lib/bindings/python/rust/engine.rs

```diff
@@ -276,8 +282,16 @@ where
 {
     let item = item.map_err(|e| {
         println!();
-        Python::with_gil(|py| e.display(py));
-        ResponseProcessingError::PythonException(e.to_string())
+        let mut is_py_generator_exit = false;
+        Python::with_gil(|py| {
+            e.display(py);
+            is_py_generator_exit = e.is_instance_of::<pyo3::exceptions::PyGeneratorExit>(py);
+        });
+        if is_py_generator_exit {
+            ResponseProcessingError::PyGeneratorExit(e.to_string())
+        } else {
+            ResponseProcessingError::PythonException(e.to_string())
```

Comment:
Just curious - I don't see any code uniquely catching PyGeneratorExit compared to PythonException other than modifying the error string. 

How is the new unique type PyGeneratorExit being used to distinguish from PythonException in terms of enabling request migration?

Reply:
Oh - I missed the error string comparison - nevermind.